src/noisy_graph_states/tools/patterns.py

```diff
@@ -0,0 +1,231 @@
+    > "Imperfect quantum networks with tailored resource states."
+    > arXiv preprint arXiv:2403.19778 (2024).
+
+    Parameters
+    ----------
+    sequence : list[tuple]
+        A series of graph measurements that are performed in order.
+        Format: ("x" or "y" or "z", qubit_index) for x measurements
+        optionally one can specify the index of the special neighbour
+        like so ("x", qubit_index, b0).
+    graph : gg.Graph
+        The start graph to which the sequence of measurements is applied.
+
+
+    Returns
+    -------
+    list[str]
+        The measurement pattern for postponed correction operations from
+        the lowest qubit index to the highest qubit index.
+        "x", "y" or "z" for a Pauli measurement. "." for no measurement.
+    """
+    matchings = defaultdict(_default_projector_matching)
+    pattern = ["."] * graph.N
+    for instruction in sequence:
+        instruction_type = instruction[0]
+        qubit_index = instruction[1]
+        pattern[qubit_index] = matchings[qubit_index][instruction_type]
+        if instruction_type == "z":
+            # no change in projector matchings required
+            graph = gt.measure_z(graph=graph, index=qubit_index)
+        elif instruction_type == "y":
+            neighbours = gt.neighbourhood(graph=graph, index=qubit_index)
+            for neighbour in neighbours:
+                match = matchings[neighbour]
+                tmp = match["x"]
+                match["x"] = match["y"]
+                match["y"] = tmp
+            graph = gt.measure_y(graph=graph, index=qubit_index)
+        elif instruction_type == "x":
+            try:
+                b0 = instruction[2]
+            except IndexError:
+                b0 = None
+            if b0 is None:
+                neighbours = gt.neighbourhood(graph=graph, index=qubit_index)
+                try:
+                    b0 = neighbours[0]
+                except IndexError:
+                    b0 = None
+                if b0 is not None:
+                    match = matchings[b0]
+                    tmp = match["x"]
+                    match["x"] = match["z"]
+                    match["z"] = tmp
+            graph = gt.measure_x(graph=graph, index=qubit_index, b0=b0)
+    return pattern
+
+
+def pattern_to_sequence(pattern: list[str], graph: gg.Graph):
+    """Find a sequence of graph measurements corresponding to a measurement pattern.
+
+    Since multiple sequences can lead to the same pattern, this
+    just returns one particular one (the one in which the graph
+    transformations are from the lowest qubit index to the
+    highest qubit index in order).
+    Use `pattern_to_all_sequences` instead to get a list of all.
+
+    This is explained in the appendix of
+    > Mor-Ruiz, Maria Flors, Julius Wallnöfer, and Wolfgang Dür.
+    > "Imperfect quantum networks with tailored resource states."
+    > arXiv preprint arXiv:2403.19778 (2024).
```

Comment:
```suggestion
    | This is explained in the appendix of
    | Maria Flors Mor-Ruiz, Julius Wallnöfer, and Wolfgang Dür.
    | "Imperfect quantum networks with tailored resource states."
    | arXiv preprint arXiv:2403.19778 (2024).
```
This formatting will make it look nice in the docs.

Also should all be the same first-name last-name order?